### PR TITLE
Add tests to #4726 docker environments and variables

### PR DESCRIPTION
**Pull request by stefwalter**
This adds a commit and squashes the changes by @vanloswang

### CODE REVIEW

**Comment by stefwalter** [comment]
Test needs another fix

**Comment by stefwalter** [comment]
This should fix the test problems.

**Comment by stefwalter** [comment]
Fixed race in the docker container logs.

**Comment by petervo** [comment]
Marking as needswork because it has WIP commits

File: pkg/docker/run.js [hunk 1]
```diff
@@ -184,6 +214,43 @@ define([
...
+            if (volumebinding.children().length > 0) {
+                $('#mount-volumes').prop('checked', true);
+                /* make sure the volumes are visible */
+                volumebinding.show();
+            } else {
+                $('#mount-volumes').prop('checked', false);
+            }
+
+            /* delete any old env var claiming entries */
+            var envvarclaiming = $('#select-claimed-envvars');
```
Comment (dperpeet) [inline pkg/docker/run.js]: variable names should be either camelCase or old_style

File: pkg/docker/run.js [hunk 2]
```diff
@@ -184,6 +214,43 @@ define([
                 $('#expose-ports').prop('checked', false);
             }
 
+            /* delete any old volume binding entries */
+            var volumebinding = $('#select-mounted-volumes');
```
Comment (dperpeet) [inline pkg/docker/run.js]: variable names should be either camelCase or old_style

File: test/verify/check-docker [hunk 3]
```diff
@@ -261,6 +261,77 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+
+        self.login_and_go("/docker")
+        b.wait_in_text("#containers-images", "busybox:latest")
+
+        # Select to view all running containers
+        b.wait_visible("#containers-containers-filter button")
+        b.click("#containers-containers-filter button")
+        b.click("#containers-containers-filter li a[value='all']")
+
+        # Create an updated image, expose a port
```
Comment (dperpeet) [inline test/verify/check-docker]: old comment, I suggest:
`# Create an updated image, add environment variable`

File: test/verify/check-docker [hunk 4]
```diff
@@ -261,6 +261,77 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+        m.execute("mkdir /var/tmp/container-probe")
+        m.upload(["verify/files/listen-on-port.sh"], "/var/tmp/container-probe")
```
Comment (dperpeet) [inline test/verify/check-docker]: do we need that script for this container?

File: test/verify/check-docker [hunk 5]
```diff
@@ -261,6 +261,77 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+' > /var/tmp/container-probe/Dockerfile""")
+        image_name = "test/container-probe"
+        m.execute("docker build -t %s /var/tmp/container-probe" % (image_name))
+        m.execute("rm -rf /var/tmp/container-probe")
+
+        # Wait for it to appear
+        b.wait_visible("#containers-images")
+        b.wait_in_text("#containers-images", image_name)
+
+        # Run it and expose additional port via the ui
```
Comment (dperpeet) [inline test/verify/check-docker]: old comment, I suggest:
`# Run it and add another environment variable via the ui`

File: test/verify/check-docker [hunk 6]
```diff
@@ -261,6 +261,77 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+        b.wait_visible("#container-details")
+
+        # Make sure they've started
+        b.wait_in_text("#container-details", "zero='GGG'")
+        b.wait_in_text("#container-details", "SECOND='marmalade'")
+        b.wait_in_text("#container-details", "dripping.txt")
+
+        self.allow_journal_messages('.*denied.*name_connect.*docker.*')
+
+
```
Comment (dperpeet) [inline test/verify/check-docker]: extra newline

**Comment by dperpeet** [comment]
also the new test keeps failing on rhel-7

**Comment by stefwalter** [comment]
I can reproduce failure on rhel-7

File: pkg/docker/run.js (hunk 1 above)
Comment (stefwalter) [inline pkg/docker/run.js]: Fixing.

File: pkg/docker/run.js (hunk 2 above)
Comment (stefwalter) [inline pkg/docker/run.js]: Fixing.

File: test/verify/check-docker (hunk 3 above)
Comment (stefwalter) [inline test/verify/check-docker]: Fixing

File: test/verify/check-docker (hunk 4 above)
Comment (stefwalter) [inline test/verify/check-docker]: Removed.

File: test/verify/check-docker (hunk 5 above)
Comment (stefwalter) [inline test/verify/check-docker]: Fixing.

File: test/verify/check-docker (hunk 6 above)
Comment (stefwalter) [inline test/verify/check-docker]: Removed.

**Comment by dperpeet** [comment]
looks good

**Comment by dperpeet** [comment]
It's an issue again with the last line of the logs being lost.

**Comment by stefwalter** [comment]
Hopefully this fixes the issue with the last line being lost.

**Pull request closed without merging**